### PR TITLE
[IMP] expense: default expense category on scanned/uploaded expense

### PR DESCRIPTION
**Pull request by yosuanicolaus**
Description of the issue/feature this PR addresses:
- Expenses are automatically created when uploading or scanning receipts. This is a feature often used by employees who have a lot of tickets and upload them in batch. The expenses created have a random category (product_id) specified. Companies may however want to specify the default category for accounting or management purposes. They are currently unable to do so.

Current behavior before PR:
- Companies can't specify default category for scanned/uploaded expenses

Desired behavior after PR is merged:
- In the Expenses setting, combine the "Default Journals" and "Payment" block into one and name the new block "Accounting"
- Add new setting in Expenses for specifying default category for scanned/uploaded expenses
- If that setting is set, use the value inside it for the category field of new scanned/uploaded expenses

Task-id: 3354763

### CODE REVIEW

File: addons/hr_expense/models/hr_expense.py [hunk 1]
```diff
@@ -347,13 +347,14 @@ def create_expense_from_attachments(self, attachment_ids=None, view_type='list')
             product = product.filtered(lambda p: p.default_code == "EXP_GEN")[:1] or product[0]
         else:
             raise UserError(_("You need to have at least one category that can be expensed in your database to proceed!"))
+        default_category = self.env.company.expense_category_id
```
Comment (rugo-odoo) [inline addons/hr_expense/models/hr_expense.py:350]: I was gonna ask you to rename this variable `default_expense_category` but I think it's actually better to just get rid of it. I don't think it adds much more readability.

**Review by rugo-odoo — CHANGES_REQUESTED** [review]

File: addons/hr_expense/models/res_company.py [hunk 2]
```diff
@@ -6,6 +6,12 @@
 class ResCompany(models.Model):
     _inherit = "res.company"
 
+    expense_category_id = fields.Many2one(
+        "product.product",
+        string="Default Expense Category",
+        check_company=True,
+        domain="[('can_be_expensed', '=', True), '|', ('company_id', '=', False), ('company_id', '=', company_id)]",
+    )
```
Comment (qdp-odoo) [inline addons/hr_expense/models/res_company.py:14]: field name is confusing: sounds like a product.category

You should rename that into expense_product_id

**Review by qdp-odoo — CHANGES_REQUESTED** [review]
made a review

File: addons/hr_expense/models/res_company.py (hunk 2 above)
Comment (yosuanicolaus) [reply addons/hr_expense/models/res_company.py:14]: okok

**Comment by qdp-odoo** [comment]
@robodoo r+ 
@yosuanicolaus thanks, good job :-)

**Pull request closed without merging**